### DISCUSSION TITLE
MIDI CC implementation

### DISCUSSION THREAD

**Discussion by toppasaurus** [ppy/osu#18079] [Ideas]
As mentioned in my last post, I recently set up my inputs with a MIDI pad controller. I tried and failed to set the volume adjust to one of the knobs. I understand if it's too much of a hassle to implement, but being able to map MIDI CCs to adjustable values would be super cool.

The values in particular I would like to be able to adjust are the volume levels, but being able to control replay speed would be cool as well. Maybe also adjusting whatever slider the cursor is hovering over, such as DT speed or the AR in difficulty adjust.

I'm not a mapper, but I imagine there would be some nice QoL stuff there as well.

**Comment by bdach** [comment]
MIDI event parsing is [rudimentary and best-effort right now](https://github.com/ppy/osu-framework/blob/7698fd35daadb0683aa7435b03e836ee97184802/osu.Framework/Input/Handlers/Midi/MidiHandler.cs#L142-L145). Probably not going to be expanded without external development effort/help as we have much more stuff to do when it comes to stuff people actually tend to use often (read: PRs welcome).

**Comment by peppy** [comment]
I don't see a reason not to add support for this, but it is very niche so unless someone offers to implement it (or there is considerable demand) it may not happen.
